dot/state/test_helpers.go

```diff
@@ -240,6 +240,8 @@ func generateBlockWithRandomTrie(t *testing.T, serv *Service,
 	trieState, err := serv.Storage.TrieState(nil)
 	require.NoError(t, err)
 
+	trieState = trieState.Snapshot()
```

Comment:
again why do we need to snapshot here?

Reply:
See https://github.com/ChainSafe/gossamer/pull/2912#discussion_r1085494292